src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.QueueItem.cs

```diff
@@ -5,59 +5,79 @@
 #nullable enable
 
 using System;
-using System.Collections.Immutable;
 using System.Diagnostics;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler
 {
     internal partial class RequestExecutionQueue
     {
-        private readonly struct QueueItem
+        private interface IQueueItem
         {
             /// <summary>
-            /// Callback to call into underlying <see cref="IRequestHandler"/> to perform the actual work of this item.
+            /// Begins executing the work specified by this queue item.
             /// </summary>
-            private readonly Func<RequestContext?, CancellationToken, Task> _callbackAsync;
+            public Task CallbackAsync(RequestContext? context, CancellationToken cancellationToken);
 
-            /// <summary>
-            /// <see cref="CorrelationManager.ActivityId"/> used to properly correlate this work with the loghub
-            /// tracing/logging subsystem.
-            /// </summary>
-            public readonly Guid ActivityId;
-            private readonly ILspLogger _logger;
+            /// <inheritdoc cref="IRequestHandler.RequiresLSPSolution" />
+            public bool RequiresLSPSolution { get; }
 
             /// <inheritdoc cref="IRequestHandler.MutatesSolutionState" />
-            public readonly bool MutatesSolutionState;
-
-            /// <inheritdoc cref="IRequestHandler.RequiresLSPSolution" />
-            public readonly bool RequiresLSPSolution;
+            public bool MutatesSolutionState { get; }
 
             /// <inheritdoc cref="RequestContext.ClientName" />
-            public readonly string? ClientName;
-            public readonly string MethodName;
+            public string? ClientName { get; }
 
-            /// <inheritdoc cref="RequestContext.ClientCapabilities" />
-            public readonly ClientCapabilities ClientCapabilities;
+            public string MethodName { get; }
 
             /// <summary>
             /// The document identifier that will be used to find the solution and document for this request. This comes from the <see cref="TextDocumentIdentifier"/> returned from the handler itself via a call to <see cref="IRequestHandler{RequestType, ResponseType}.GetTextDocumentIdentifier(RequestType)"/>.
             /// </summary>
-            public readonly TextDocumentIdentifier? TextDocument;
+            public TextDocumentIdentifier? TextDocument { get; }
+
+            /// <inheritdoc cref="RequestContext.ClientCapabilities" />
+            public ClientCapabilities ClientCapabilities { get; }
```

Comment:
just to check, but do these all need to be public?  specifically, is there a consumer of each of these through teh interface?  if so, cool.  if not, def reduce surface area if possible.